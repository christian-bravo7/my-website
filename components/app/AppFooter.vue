<template>
  <footer class="py-16 px-4 bg-gray-800 dark:bg-gray-900">
    <div class="mx-auto container flex justify-center items-center flex-col">
      <div class="grid grid-cols-4 gap-x-6 mb-10">
        <FooterSocialIcon link="https://codepen.io/christian-bravo">
          <CodepenIcon />
        </FooterSocialIcon>
        <FooterSocialIcon link="https://github.com/christian-bravo7">
          <GithubIcon />
        </FooterSocialIcon>
        <FooterSocialIcon link="https://www.instagram.com/christian.bravo_/">
          <InstagramIcon />
        </FooterSocialIcon>
        <FooterSocialIcon link="https://www.linkedin.com/in/christian151-bravo/">
          <LinkedinIcon />
        </FooterSocialIcon>
      </div>
      <p class="text-white text-xs">
        {{ $t('me.crafted-label') }} <span
          role="img"
          aria-label="heart"
        >❤️</span> | &copy; Christian Bravo
      </p>
    </div>
  </footer>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

import CodepenIcon from '~/assets/img/codepen.svg?inline';
import GithubIcon from '~/assets/img/github.svg?inline';
import InstagramIcon from '~/assets/img/instagram.svg?inline';
import LinkedinIcon from '~/assets/img/linkedin.svg?inline';

@Component({
  components: {
    CodepenIcon,
    GithubIcon,
    InstagramIcon,
    LinkedinIcon,
  },
})
export default class AppFooter extends Vue {}

</script>
