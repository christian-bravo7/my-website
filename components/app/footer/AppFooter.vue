<template>
  <footer class="py-10 px-4 | bg-gray-100 dark:bg-blue-900 | text-black dark:text-white rounded-tl-3xl rounded-tr-3xl">
    <div class="container | my-0 mx-auto">
      <div class="flex justify-between flex-col md:flex-row">
        <div class="flex items-center flex-col md:flex-row | mb-6 md:mb-0">
          <div class="mr-0 mb-4 md:mr-8 md:mb-0">
            <AppLogo class="w-10 h-10 md:w-12 md:h-12" />
          </div>
          <div class="flex flex-col text-center md:text-left">
            <h3 class="text-lg">
              Christian Bravo
            </h3>
            <h4 class="text-sm">
              {{ $t('footer.frontend-developer') }}
            </h4>
          </div>
        </div>
        <div class="flex flex-col justify-center items-center md:items-end | text-xs | text-right">
          <span>
            Â© 2021 Christian Bravo
          </span>
          <span>
            {{ $t('footer.made-with-love') }}
          </span>
        </div>
      </div>
    </div>
  </footer>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

@Component
export default class AppFooter extends Vue {}
</script>
