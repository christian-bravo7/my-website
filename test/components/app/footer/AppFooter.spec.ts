import { createLocalVue, mount } from '@vue/test-utils';

import AppFooter from '@/components/app/footer/AppFooter.vue';

import setupI18n from '@/test/i18n';

const localVue = createLocalVue();
const i18n = setupI18n(localVue);

const componentConfig = {
  localVue,
  i18n,
};

describe('AppFooter component', () => {
  it('should render correclty', () => {
    const wrapper = mount(AppFooter, {
      ...componentConfig,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
