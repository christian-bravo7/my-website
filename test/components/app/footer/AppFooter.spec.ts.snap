// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter component should render correclty 1`] = `
<footer class="py-10 px-4 | bg-gray-100 dark:bg-blue-900 | text-gray-700 dark:text-gray-300">
  <div class="container | my-0 mx-auto">
    <div class="flex justify-between flex-col md:flex-row">
      <div class="flex items-center flex-col md:flex-row | mb-6 md:mb-0">
        <div class="mr-0 mb-4 md:mr-8 md:mb-0"><a class="flex col-span-1 | w-8 h-8 | text-black dark:text-white w-10 h-10 md:w-12 md:h-12"><svg class="w-full h-full | fill-current"></svg></a></div>
        <div class="flex flex-col | text-center md:text-left | text-base md:text-lg">
          <h3>
            Christian Bravo
          </h3>
          <h4>
            Frontend developer
          </h4>
        </div>
      </div>
      <div class="flex flex-col justify-center items-center md:items-end | text-xs | text-right"><span>
          Â© 2021 Christian Bravo
        </span> <span>
          Made with Love and Nuxt.js
        </span></div>
    </div>
  </div>
</footer>
`;
